packages/reactive-dict/reactive-dict.js

```diff
@@ -1,117 +1,114 @@
 // XXX come up with a serialization method which canonicalizes object key
 // order, which would allow us to use objects as values for equals.
-var stringify = function (value) {
-  if (value === undefined)
+function stringify(value) {
+  if (value === undefined) {
     return 'undefined';
+  }
   return EJSON.stringify(value);
-};
-var parse = function (serialized) {
-  if (serialized === undefined || serialized === 'undefined')
+}
+
+function parse(serialized) {
+  if (serialized === undefined || serialized === 'undefined') {
     return undefined;
+  }
   return EJSON.parse(serialized);
-};
+}
 
-var changed = function (v) {
+function changed(v) {
   v && v.changed();
-};
+}
 
 // XXX COMPAT WITH 0.9.1 : accept migrationData instead of dictName
-ReactiveDict = function (dictName) {
-  // this.keys: key -> value
-  if (dictName) {
-    if (typeof dictName === 'string') {
-      // the normal case, argument is a string name.
-      // _registerDictForMigrate will throw an error on duplicate name.
-      ReactiveDict._registerDictForMigrate(dictName, this);
-      this.keys = ReactiveDict._loadMigratedDict(dictName) || {};
-      this.name = dictName;
-    } else if (typeof dictName === 'object') {
-      // back-compat case: dictName is actually migrationData
-      this.keys = dictName;
-    } else {
-      throw new Error("Invalid ReactiveDict argument: " + dictName);
-    }
-  } else {
-    // no name given; no migration will be performed
+export class ReactiveDict {
+  constructor(dictName, dictData = {}) {
+    // this.keys: key -> value
     this.keys = {};
-  }
 
-  this.allDeps = new Tracker.Dependency;
-  this.keyDeps = {}; // key -> Dependency
-  this.keyValueDeps = {}; // key -> Dependency
-};
+    if (dictName) {
```

Comment:
I've looked into this a bit and as we have support for `new ReactiveDict({ property: 'some value'});` I think we should throw an error for this `new ReactiveDict(null/undefined, { property: 'some value' });`.

Even though `new ReactiveDict({ property: 'some value' });` was there initially to support some old kind of migration data it isn't really tied to this in some way.

However thinking about this case.
```js
let dictName;

const dict = new ReactiveDict(dictName, { property: 'some value' });
```
In this case the user might be surprised that the dict isn't migrated on reload because we created an unnamed dict instead.

However, taking into account that we don't care about this in the case when we don't provide initial data it might be a bad idea.